### PR TITLE
doc: update compile-time configuration HLD

### PR DESCRIPTION
**Pull request by junjiemao1**
Starting from v2.4, ACRN configuration uses solely XML files to store
configuration data and customized scripts to manipulate
configurations. This patch updates the HLD of compile-time configuration to
reflect this properly.

As the refinement to the configuration toolset itself is still ongoing,
this patch only adds brief introduction to the key generated files involved
in ACRN configuration. More details will be added after the refinement
completes.

Tracked-On: #5644
Signed-off-by: Junjie Mao <junjie.mao@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hv-config.rst [hunk 1]
```diff
@@ -3,54 +3,44 @@
...
-
-   c. **menuconfig** presents a terminal UI to navigate and modify the
-      configurations in an interactive manner.
-
-2. The target oldconfig is also used to create a .config if a .config
-   file does not exist when building the source directly.
-
-3. The other two files for makefiles and C sources are regenerated after
-   .config changes.
+Whenever ``.board.xml`` or ``.scenario.xml`` is modified, the hypervisor will be
```
Comment (gvancuts) [inline doc/developer-guides/hld/hv-config.rst:41]: Is this the way to modify an existing build?

What happens if someone modifies other files such as `configs/config.mk` or others?

**Review by gvancuts — APPROVED** [review]
LGTM!

**Review by fitchbe — APPROVED** [review]
Pushed a few copy edits.

File: doc/developer-guides/hld/hv-config.rst (hunk 1 above)
Comment (junjiemao1) [reply doc/developer-guides/hld/hv-config.rst:41]: Yes, modifying `.board.xml` and `.scenario.xml` is the only supported way to modify the configurations of an existing build. Changing other generated files (e.g. `config.h`, `config.mk` or else) will lead to unpredictable behavior, possibly due to inconsistent data in these files.

In v2.5 the preferred way will be using the configuration editor (via the `menuconfig` target) We still need to add a few command line options to the editor to enable this.

**Pull request merged**